### PR TITLE
Remove tron/utils/collections

### PR DESCRIPTION
**Pull request by keymone**
- remove backported Enum class
- remove MappingCollection
- also split up dev requirements from doc requirements

### CODE REVIEW

File: tron/core/job.py [hunk 1]
```diff
@@ -435,32 +456,37 @@ class JobCollection(object):
     """A collection of jobs."""
 
     def __init__(self):
-        self.jobs = collections.MappingCollection('jobs')
+        self.jobs = {}
```
Comment (qui) [inline tron/core/job.py]: There are other references to self.jobs in this class, do you need to update those too?

File: tox.ini [hunk 2]
```diff
@@ -19,7 +19,7 @@ ignore = E501,E265,E241,E704
 
 [testenv:docs]
 deps =
-  --requirement={toxinidir}/dev/req_dev.txt
+  --requirement={toxinidir}/dev/req_docs.txt
```
Comment (solarkennedy) [inline tox.ini]: I'm surprised this works without req_dev.txt too. Can you confirm it actually does work as written?

Comment (keymone) [reply tox.ini]: can confirm, make docs passes provided services code is cleaned up

File: tron/core/job.py (hunk 1 above)
Comment (keymone) [reply tron/core/job.py]: i've checked - they all use dict-interface, so we're fine

**Pull request merged**